internal/util/debug/debug.go

```diff
@@ -49,6 +49,11 @@ const (
 // setup ensures that debug handler is set up only once.
 var setup atomic.Bool
 
+// Probe should return true on success and false on failure.
+//
+// It must be thread-safe.
+type Probe func() bool
```

Comment:
Actually, when I think about it now. I think the function should include context as an argument.

The readiness probe has some operations behind it, so it'd be useful to have Cancellation option based on the request context.